pkg/asset/installconfig/azure/session.go

```diff
@@ -35,18 +35,25 @@ type Credentials struct {
 // GetSession returns an azure session by using credentials found in ~/.azure/osServicePrincipal.json
 // and, if no creds are found, asks for them and stores them on disk in a config file
 func GetSession() (*Session, error) {
-	os.Setenv(azureAuthEnv, authFilePath)
-	return newSessionFromFile()
+	authFile := defaultAuthFilePath
+	if f, ok := os.LookupEnv(azureAuthEnv); ok && len(f) > 0 {
```

Comment:
If you're going to gate on length anyway, you might as well just use [`os.Getenv`][1].

[1]: https://golang.org/pkg/os/#Getenv

Reply:
fixed with https://github.com/openshift/installer/compare/4bff9ab1f5f879ccdabd9fa5730d21b711aec476...0e9453e723a95ec3a6290e2985d7926f8a3f7618